packages/nc-gui/components/smartsheet/column/FormulaOptions.vue

```diff
@@ -231,6 +231,40 @@ function validateAgainstMeta(parsedTree: any, errors = new Set(), typeErrors = n
             },
             typeErrors,
           )
+        } else if (parsedTree.callee.name === 'DATETIME_DIFF') {
+          // parsedTree.arguments[0] = date
+          validateAgainstType(
+            parsedTree.arguments[0],
+            formulaTypes.DATE,
+            (v: any) => {
+              if (!validateDateWithUnknownFormat(v)) {
+                typeErrors.add('The first parameter of DATEADD() should have date value')
+              }
+            },
+            typeErrors,
+          )
+          // parsedTree.arguments[1] = date
+          validateAgainstType(
+            parsedTree.arguments[1],
+            formulaTypes.DATE,
+            (v: any) => {
+              if (!validateDateWithUnknownFormat(v)) {
+                typeErrors.add('The second parameter of DATEADD() should have date value')
+              }
+            },
+            typeErrors,
+          )
+           // parsedTree.arguments[2] = ["millisecond" | "ms" | "second" | "s" | "minute" | "mi" | "hour" | "hh" | "day" | "dy" | "week" | "wk" | "month" | "m" | "quarter" | "q" | "year" | "y"]
+           validateAgainstType(
+            parsedTree.arguments[2],
+            formulaTypes.STRING,
+            (v: any) => {
+              if (!['millisecond', 'ms', 'second', 's', 'minute', 'mi', 'hour', 'hh', 'day', 'dy', 'week', 'wk', 'month', 'm', 'quarter', 'q', 'year', 'y'].includes(v)) {
+                typeErrors.add('The third parameter of DATEADD() should have value either "millisecond", "ms", "second", "s", "minute", "mi", "hour", "hh", "day", "dy", "week", "wk", "month", "m", "quarter", "q", "year", or "y"')
+              }
+            },
+            typeErrors,
+          )
```

Comment:
I guess I'm confused because this seems to be the correct function for this task: https://www.w3schools.com/sql/func_sqlserver_datediff.asp

Also, I changed the list provided to match the accepted list for datediff for consistency.

```
year, yyyy, yy = Year
quarter, qq, q = Quarter
month, mm, m = month
dayofyear = Day of the year
day, dy, y = Day
week, ww, wk = Week
weekday, dw, w = Weekday
hour, hh = hour
minute, mi, n = Minute
second, ss, s = Second
millisecond, ms = Millisecond
```

My SQL knowledge could use some growth so if there is another function that would work better please let me know. Thanks for your feedback.

Reply:
I think I see what you're saying now does this look like the correct function?
https://www.w3schools.com/mysql/trymysql.asp?filename=trysql_func_mysql_datediff

Reply:
1. The list I provided is retrieved from Airtable, as we want to minimize the conversion during the airtable migration and also AT users don't need to learn the same formula with different arguments.
2. For each DB, you need to find the correct function. The one you used is for MSSQL, not for MySQL. We couldn't feed `unit` to `DATEDIFF`. I believe the correct one is `TIMESTAMPDIFF`. The syntax is `TIMESTAMPDIFF(unit, datetime_expr1, datetime_expr2)`. You may try it.

Reply:
Ok, I understand why those values need to be used now I'll write a helper function to convert them to the correct ones per language. Thanks for pointing me in the right direction.